### PR TITLE
Adds grunt-contrib-concat

### PR DESCRIPTION
**Pull request by lukecampbell**
- Adds task to concatenate all the javascript modules for science UI
   into one JS.
 - We'll need to minify this for production's sake.

Note: d3.js won't work and will need to be explicitly sourced because
it's not strictly compliant.

### CODE REVIEW

**Comment by lukecampbell** [comment]
@birdage @oceanzus please review

File: ooiui/static/js/core/science/map.js [hunk 1]
```diff
@@ -65,7 +65,7 @@ function getStatusMarker(status,stream_num){
 function addMarkers(map,markers){
 
     /* PLATFORM MARKER EXTENDED FROM NORMAL MARKER*/
-    PlatformMarker = L.Marker.extend({
+    var PlatformMarker = L.Marker.extend({
```
Comment (lukecampbell) [inline ooiui/static/js/core/science/map.js]: `use strict` caught this.

File: Gruntfile.js [hunk 2]
```diff
@@ -2,10 +2,34 @@ module.exports = function(grunt) {
...
+      dist: {
+        options: {
+          banner: "'use strict';\n",
+          process: function(src, filepath) {
+            return '// Source: ' + filepath + '\n' +
+              src.replace(/(^|\n)[ \t]*('use strict'|"use strict");?\s*/g, '$1');
+          },
+        },
+        files: {
+          'ooiui/static/js/compiled/science.js' : [
```
Comment (lukecampbell) [inline Gruntfile.js]: so when we build out layouts and pages we can just specify the dependent JS files here. This replaces us having to explicitly import them and/or use require.js

Once we minify them, it'll be way faster too.

**Comment by lukecampbell** [comment]
@birdage please review and merge at your earliest leisure.

**Comment by lukecampbell** [comment]
Once we move to your new index, a lot of the dependent JS files will get removed, I think.

**Comment by birdage** [comment]
@lukecampbell indeed they should

**Pull request merged**